source/updateCheck.py

```diff
@@ -245,32 +307,84 @@ def onDownloadButton(self, evt):
 		DonateRequestDialog(gui.mainFrame, self._download)
 
 	def _download(self):
-		if self.isInstalled:
-			UpdateDownloader(self.urls, fileHash=self.fileHash).start()
-		else:
-			os.startfile(self.urls[0])
+		UpdateDownloader(self.updateInfo).start()
 		self.Destroy()
 
 	def onLaterButton(self, evt):
 		state["dontRemindVersion"] = None
 		saveState()
 		self.Close()
 
+class UpdateAskInstallDialog(wx.Dialog):
+
+	def __init__(self, parent, destPath, version):
+		self.destPath=destPath
+		self.version = version
+		storeUpdatesDirWritable=os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
+		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		super(UpdateAskInstallDialog, self).__init__(parent, title=_("NVDA Update"))
+		mainSizer = wx.BoxSizer(wx.VERTICAL)
+		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
+		
+		# Translators: A message indicating that an updated version of NVDA is ready to be installed.
+		sHelper.addItem(wx.StaticText(self, label=_("Update is ready to install.\n")))

```

Comment:
Do you think this dialog should tell you the updates version information? To be consistent with the other download update dialog?